feature/component-filter-bottomsheet

Mostafa-alsaygh opened the pull request:
This branch introduces the filter bottom sheet feature, allowing users to filter content based on release year, genre, and IMDb rating.
<img width="554" height="653" alt="image" src="https://github.com/user-attachments/assets/8f915971-0123-47f0-a718-a73441b52c15" />

Mostafa-alsaygh committed:
add a filter bottom sheet component

Mostafa-alsaygh committed:
Merge remote-tracking branch 'origin/develop' into feature/component-filter-bottomsheet

Mostafa-alsaygh committed:
feat: implement filter bottom sheet functionality

This commit introduces the filter bottom sheet feature, allowing users to filter content based on release year, genre, and IMDb rating.

Key changes include:
- Moving `FilterBottomSheet` and `FilterBottomSheetUiState` from `designSystem` to `presentation`.
- Updating `FilterBottomSheetUiState` with new fields: `selectedGenre`, `imdbRating`, and `yearRange`.
- Adding new composable components:
    - `CustomReleasedYearSlider`: A slider for selecting a range of release years.
    - `GenreChipGroup`: A group of chips for selecting genres.
    - `RatingBar`: A component for selecting IMDb rating.
- Implementing the logic within `FilterBottomSheet` to handle user selections and apply/clear filters.
- Adding new drawable resources for star icons (filled and outline).
- Adding string resources for filter-related UI elements in English and Arabic.
- Removing outdated string resources from `designSystem`.

Mostafa-alsaygh committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,103 @@
+                .padding(top = 8.dp),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(text = minYear.toString(), color = Color.Gray)
```
use our theme colors in the design system NovixTheme.colors.body

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,103 @@
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(text = minYear.toString(), color = Color.Gray)
+            Text(text = maxYear.toString(), color = Color.Gray)
```
also here

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,103 @@
+                        .background(NovixTheme.colors.surface)
+                        .border(
+                            width = 1.dp,
+                            color = NovixTheme.colors.stroke, // light gray border
```
delete unnecessary comments

Mostafa-alsaygh committed:
fix: adjust FilterBottomSheet styles and typography

- Updated the `FilterBottomSheet` drag handle color to use `NovixTheme.colors.body`.
- Removed `rememberSaveable` from `yearRange` in `FilterBottomSheetContent` as it was causing a crash.
- Removed bottom padding from the main column in `FilterBottomSheetContent`.
- Adjusted bottom padding for the "Released Year" text.
- Modified title typography sizes and line heights in `NovixThemeTypography`.
- Updated text colors and styles in `CustomReleasedYearSlider` to use `NovixTheme` values.
- Capitalized "IMDB rating" string resource.

Mostafa-alsaygh committed:
refactor: adjust rounded corners in CustomReleasedYearSlider

Mostafa-alsaygh committed:
style: update border radius of custom released year slider

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,111 @@
+                                (startFraction * (LocalConfiguration.current.screenWidthDp - 32).dp.toPx()).toDp()
+                            })
+                            .height(8.dp)
+                            .background(Color(0xFFF2674A))
```
get the colors from our design system

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,111 @@
+                Box(
+                    modifier = Modifier
+                        .size(20.dp)
+                        .background(Color(0xFFF2674A), shape = CircleShape)
```
also here

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,111 @@
+                Box(
+                    modifier = Modifier
+                        .size(20.dp)
+                        .background(Color(0xFFF2674A), shape = CircleShape)
```
and here

Mostafa-alsaygh committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/CustomReleasedYearSlider.kt:
```diff
@@ -0,0 +1,111 @@
+        )
+    }
+}
+
```
add a preview to show your work, for example:

@ThemePreviews
@Composable
fun CustomReleasedYearSliderPreview() {
    NovixTheme {
        CustomReleasedYearSlider(
            yearRange = 2000f..2023f,
            onYearRangeChange = {}
        )
    }
}

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/RatingBar.kt:
```diff
@@ -0,0 +1,65 @@
+        modifier = modifier,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
```
space padding between stars is equal to 8 not 4

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/RatingBar.kt:
```diff
@@ -0,0 +1,65 @@
+                        painterResource(R.drawable.star_outline),
+                    contentDescription = "Rate $i",
+                    tint = NovixTheme.colors.yellowAccent,
+                    modifier = Modifier.padding(end = 8.dp)
```
make horizontal padding instead of end padding

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/composables/filterbottomsheet/FilterBottomSheet.kt:
```diff
@@ -0,0 +1,252 @@
+                .fillMaxWidth()
+                .align(Alignment.End)
+        ) {
+            Box(
```
the buttons component is already in the components package no need to make a custom buttons again

Mostafa-alsaygh committed:
Fix: refactor and improve filter bottom sheet and components

Mostafa-alsaygh committed:
feat: Enable R8 full mode for release builds

Mostafa-alsaygh committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

# Conflicts:
#	designSystem/src/main/res/values-ar/strings.xml
#	designSystem/src/main/res/values/strings.xml
#	presentation/src/main/res/values-ar/strings.xml
#	presentation/src/main/res/values/strings.xml

BassantM committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

Commit:
Merge branch 'develop' into feature/component-filter-bottomsheet

Mostafa-alsaygh committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

ahmeddsamii reviewed (changes requested):
solve the build issue

BassantM committed:
Merge branch 'develop' into feature/component-filter-bottomsheet

BassantM committed:
feat: integrate filter bottom sheet to search screen

BassantM committed:
feat: update filter bottom sheet with view model

Mostafa-alsaygh committed:
Refactor: rename filter functions in SearchViewModel

- Renamed `applyFilterBottomSheet` to `onApplyFilter`.
- Renamed `clearFilterBottomSheet` to `onClearFilter`.
- Renamed `_uiSFilterState` to `_filterUiState`.
- Capitalized "IMDb" in Arabic "imdb_rating" string resource.

Mostafa-alsaygh committed:
Opt in to ExperimentalLayoutApi for GenreChipGroup
